### ISSUE TITLE
Networkx transitive reduction very slow

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#332]
In GitLab by @ lbatista on May 29, 2018, 13:32

For a complex workflow with FREQUENCY, SYNCHRONIZED and redundant dependencies (because of the frequency), with 25 members and 100+ chunks, the transitive reduction takes 5 minutes and it increases to 20 and 80 minutes as the number of chunks doubles.

**Comment by kinow** [comment]
In GitLab by @ editomas on Jun 22, 2018, 19:00

@ lbatista I've tested the option `autosubmit create -nt a172 -np` and when creating a 1 year experiment with auto-monarch (ENS+DA) the time is reduced from hours to seconds. Fantastic! I have not run the experiment though, to double check that the flow is created correctly, but I trust it is. 

It would be very useful if we could use the option "-nt" also with setstatus. If a long experiment crashes, this also takes a very long time (hours). Beside handling crashes, this is useful also when we extend for a long period an experiment, and don't want to re-run certain initial jobs (compilation, etc...).

**Comment by kinow** [comment]
In GitLab by @ lbatista on Jul 24, 2018, 18:12

Just a simple workflow comparing using -nt (on the right) and not using (on the left):

![Screenshot_from_2018-07-24_18-00-42](https://earth.bsc.es/gitlab/es/autosubmit/uploads/6ee28632cd82d1186ef8a3a4e98dc15f/Screenshot_from_2018-07-24_18-00-42.png)

```
[SIM]
FILE = SIM.sh
DEPENDENCIES = SIM-1
RUNNING = chunk
WALLCLOCK = 00:05
PROCESSORS = 2
THREADS = 1

[POST]
FILE = adapt.sh
DEPENDENCIES = SIM
RUNNING = chunk
FREQUENCY = 2
WALLCLOCK = 00:05

[CLEAN]
FILE = clean.sh
DEPENDENCIES = POST SIM
RUNNING = chunk
WALLCLOCK = 00:05
```


I have added this option for autosubmit check, monitor, recovery, run, set status and stats, apart from create.

**Comment by kinow** [comment]
In GitLab by @ lbatista on Jul 24, 2018, 18:27

closed via commit f438e7edd61e7e8d36550c13b2e5bb604a8e52ce

**Comment by kinow** [comment]
In GitLab by @ lbatista on Jul 24, 2018, 19:01

mentioned in commit f438e7edd61e7e8d36550c13b2e5bb604a8e52ce

**Comment by kinow** [comment]
In GitLab by @ lbatista on Jul 27, 2018, 17:32

It is important to keep in mind a side-effect of this option: while it prevents the bottleneck by eliminating the transitive reduction, the plotting step becomes much slower as there are a lot of redundant dependencies. Therefore, this is more recommended when the workflow is incredibly large and the plotting is useless (along with the -np flag). This can potentially be fixed with #333
